sgf-viewer/src/components/NavBar.svelte

```diff
@@ -1,185 +1,85 @@
 <script lang="ts">
     import { pages } from "../content";
-    import NavButtons from "./subcomponents/NavButtons.svelte";
-    import TableOfContents from "./subcomponents/TableOfContents.svelte";
-    import FaGithub from "svelte-icons/fa/FaGithub.svelte";
-    import FaFile from "svelte-icons/fa/FaFile.svelte";
-    import TiHome from "svelte-icons/ti/TiHome.svelte";
-    import IoIosMenu from "svelte-icons/io/IoIosMenu.svelte";
 
-    export let contentsFloatWidth: number;
-    export let menuNavigationWidth: number;
+    import { Collapse } from "bootstrap";
+
     export let currentPath: string;
-    export let navBarElem: HTMLElement;
 
     let pagesPaths = Object.keys(pages);
-    let innerWidth;
-    let menuOpen = false;
+    let navbarSupportedContent;
+
+    // Make the navbar collapse when a non-dropdown link is clicked on mobile.
+    // This is needed because we have a single-page app, and our links don't
+    // actually open a new page.
+    // Based on https://stackoverflow.com/a/42401686/4865149 but with
+    // modifications to make it work with Svelte's dynamic element rendering.
+    // (The other solution using HTML attributes given in the SO answer, though
+    // it's simpler, breaks the links' actual href functionality.)
+    $: navLinks = navbarSupportedContent?.querySelectorAll('.nav-item:not(.dropdown)');
+    $: bootstrapCollapse = navbarSupportedContent ? Collapse.getOrCreateInstance(navbarSupportedContent, {toggle: false}) : undefined;
+    $: navLinks?.forEach((link) => {
+           link.addEventListener('click', () => { bootstrapCollapse.toggle() });
```

Comment:
Could `bootstrapCollapse` ever be `undefined` and have `toggle()` called on it?

Reply:
no, both `navLinks` and `bootstrapCollapse` are undefined under the same conditions — when `navbarSupportedContent` is undefined